oslabs-beta/janusQL
# Implemented caching of all performance tests

## **Types of Changes**


## **Purpose**
- Cache performance metrics to display a history of all the performance tests in one session.

## **Approach**
- Instantiated a Redis server to store key-value pairs
- Incremented key per each request
- Evict cache contents after each session

## **How to Test**
- npm run dev
- send dummy url and query strings
- terminal console should display current key and associated value which is the res.locals obj

## **Screenshot(s)**
**Declared Redis server**
![redis1](https://user-images.githubusercontent.com/62489463/112351490-56920c00-8ca0-11eb-85ce-646037ecf577.png)


**Middleware function that stores performance metrics in cache**
![redis2](https://user-images.githubusercontent.com/62489463/112351515-5abe2980-8ca0-11eb-8629-e8610a672341.png)


Collaborated with @kimchiuu
